lib/cli.py

```diff
@@ -26,22 +27,40 @@ class TrainingProcessor(object):
     def __init__(self, subparser, command, description='default'):
         self.parse_arguments(description, subparser, command)
 
+        self.models = [{
+            'file': 'encoder.h5',
+            'container': encoder
+        }, {
+            'file': 'decoder_A.h5',
+            'container': decoder_A
+        }, {
+            'file': 'decoder_B.h5',
+            'container': decoder_B
+        }]
+
     def process_arguments(self, arguments):
         self.arguments = arguments
         print("Model A Directory: {}".format(self.arguments.input_A))
         print("Model B Directory: {}".format(self.arguments.input_B))
         print("Training data directory: {}".format(self.arguments.model_dir))
         print('Starting, this may take a while...')
 
-        try:
-            encoder.load_weights(self.arguments.model_dir + '/encoder.h5')
-            decoder_A.load_weights(self.arguments.model_dir + '/decoder_A.h5')
-            decoder_B.load_weights(self.arguments.model_dir + '/decoder_B.h5')
-        except Exception as e:
-            print('Not loading existing training data.')
-            print(e)
+        for model in self.models:
+            file_path = os.path.join(self.arguments.model_dir, model['file'])
+            if os.path.isfile(file_path):
+                model['container'].load_weights(file_path)
+            else:
+                print("Model file {} is not exists, creating...".format(model['file']))
+                model['container'].save_weights(file_path)
```

Comment:
Create model files before training. So script will fail almost immediately if for example there are problems with writing permissions #failfast

Reply:
This is cleaner indeed, it looks good to me